hedera-mirror-rest/service/entityService.js

```diff
@@ -66,26 +79,62 @@ class EntityService extends BaseService {
       return null;
     } else if (rows.length > 1) {
       logger.error(`Incorrect db state: ${rows.length} alive entities matching alias ${accountAlias}`);
-      throw new Error(`Multiple alive entities matching alias`);
+      throw new Error(EntityService.multipleAliasMatch);
     }
 
     return new Entity(rows[0]);
   }
 
   /**
-   * Gets the encoded account id from the account alias string. Throws {@link InvalidArgumentError} if the account alias
-   * string is invalid,
+   * Checks if provided accountId maps to a valid entity
+   * @param {String} accountId
+   * @returns {Promise} valid flag
+   */
+  async isValidAccount(accountId) {
+    const entity = await super.getSingleRow(EntityService.entityExistenceQuery, [accountId], 'isValidAccount');
+    return !_.isNil(entity);
+  }
+
+  /**
+   * Gets the encoded account id from the account alias string.
    * @param {AccountAlias} accountAlias the account alias object
    * @return {Promise}
    */
   async getAccountIdFromAlias(accountAlias) {
     const entity = await this.getAccountFromAlias(accountAlias);
     if (_.isNil(entity)) {
-      throw new NotFoundError('No account with a matching alias found');
+      throw new NotFoundError(EntityService.missingAccountAlias);
     }
 
     return entity.id;
   }
+
+  /**
+   * Retrieve the encodedId of a validated EntityId from an accountId or alias string.
+   * Throws {@link InvalidArgumentError} if the account alias string is invalid
+   * Throws {@link NotFoundError} if the account is not present
+   * @param {String} accountIdString accountIdOrAlias query string
+   * @returns {Promise} entityId
+   */
+  async getEncodedIdOfValidatedEntityId(accountIdString) {
+    let entity = null;
```

Comment:
better rename it to `encodedId` or `id` to avoid confusion.
```suggestion
    let encodedId = null;
```

Reply:
Updated